lua/telescope/actions/utils.lua

```diff
@@ -0,0 +1,58 @@
+---@tag telescope.actions.utils
+
+---@brief [[
+--- Utilities to wrap functions around picker selections and entries.
+---
+--- Generally used from within other |telescope.actions|
+---@brief ]]
+
+local action_state = require('telescope.actions.state')
+
+local utils = {}
+
+--- Apply `f` to the entries of the current picker and return a table of mapped entries.
+--- - Note: mapped entries may include results not visible in the results popup.
+---@param prompt_bufnr number: The prompt bufnr
+---@param f function: Function to apply on entries of picker that takes (entry, index, row) as possible arguments
+---@return table: Result from `f` applied to entries
+function utils.map_entries(prompt_bufnr, f)
+  vim.validate{
+    f = {f, "function"}
+  }
+  local current_picker = action_state.get_current_picker(prompt_bufnr)
+  local results = {}
+  local index = 1
+  local row, result
```

Comment:
Does this need to be outside of the loop?

Reply:
Since we seem to be focused on maximum speed (`table.insert` vs `results[index] = result`) we can also avoid re-instantiating the variable. But happy to change it to e.g .`local row = current_picker:get_row(index)` then instead.

Reply:
Yeah dont try to cheat the jit compiler. We have one. it will take care of it. `table.insert` is a exception because it can't be jit compiled. In this case i prefer the more readable code

Reply:
luajit wiki actually say dont do it. http://wiki.luajit.org/Numerical-Computing-Performance-Guide
this section. Close to the end of the page:
![image](https://user-images.githubusercontent.com/15233006/124283161-e1aa5a00-db4b-11eb-88ac-880d32f5449c.png)

Reply:
Ok will do, thanks for the reference!